libraries/autoload.rb

```diff
@@ -1,3 +1,7 @@
+unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+  raise "The compat_resources cookbook does not support chef versions older than Chef 12.0.0"
```

Comment:
In the metadata we say this requires 12.1, but we're checking for 12.0 here.  Which is it?